lib/mockstrategies/OrderedMockStrategy.js

```diff
@@ -0,0 +1,62 @@
+const expect = require('unexpected')
+  .clone()
+  .use(require('unexpected-messy'));
+const messy = require('messy');
+
+const errors = require('../errors');
+const resolveExpectedRequestProperties = require('../resolveExpectedRequestProperties');
+const trimMessyHeaders = require('../trimMessyHeaders');
+
+module.exports = class OrderedMockStrategy {
+  constructor(requestDescriptions) {
+    this.requestDescriptions = requestDescriptions;
+    this.nextRequestDescriptionIndex = 0;
+  }
+
+  get isEmpty() {
```

Comment:
This method name doesn't seem to describe what it does very well? Maybe `hasRemaining`?

Reply:
We decided to go with `isExhausting` and I've pushed the change :)